carbons/carbons_test.go

```diff
@@ -172,3 +172,58 @@ func TestUnwrap(t *testing.T) {
 		})
 	}
 }
+
+var privateValidTestCases = [...]struct {
+	outXML string
+	inXML  string
+}{
+	0: {
+		outXML: `<message xmlns="jabber:client" xmlns="jabber:client"><body xmlns="jabber:client">Neither, fair saint, if either thee dislike.</body><thread xmlns="jabber:client">0e3141cd80894871a68e6fe6b1ec56fa</thread><private xmlns="urn:xmpp:carbons:2"></private><no-copy xmlns="urn:xmpp:hints"></no-copy></message>`,
+		inXML:  `<message xmlns="jabber:client"><body>Neither, fair saint, if either thee dislike.</body><thread>0e3141cd80894871a68e6fe6b1ec56fa</thread></message>`,
+	},
+}
+
+var privateInValidTestCases = [...]struct {
```

Comment:
once the change above is made, I think these will just be valid test cases that result in no changes, so that should simplify this a little :)